doc/api/process.markdown

```diff
@@ -59,8 +59,9 @@ finished running the process will exit. Therefore you **must** only perform
 checks on the module's state (like for unit tests). The callback takes one
 argument, the code the process is exiting with.
 
-This event may not be fired if the process terminates due to signals like
-`SIGINT`, `SIGTERM`, `SIGKILL`, and `SIGHUP`.
+This event is only emitted when node exits explicitly by process.exit() or
+implicitly by the event loop draining. It will not, for example, be fired
```

Comment:
All good things to list imho. As a user, I want to know these things, and if they're not documented, how can we expect people to find this out? The only way becomes "the hard way". Which is either very time consuming trial-and-error or "the harder way", which is in production when you least want stuff to misbehave.

Also, if it's too complicated to explain to users, perhaps there's something inherently wrong with the system as it is. That may or may not be the case, I'm not making any assertions here, but if it's not the case, we should be able to document this well.